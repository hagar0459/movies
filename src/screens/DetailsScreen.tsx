/**
 * @flow
 * Created by Hagar Abdelghafar on 22.10.2021
 */

import React, { useRef } from 'react';
import
{ StyleSheet, ImageBackground, SafeAreaView, Text, View } from 'react-native';
// import { SharedElement } from 'react-navigation-shared-element';
import * as Animatable from 'react-native-animatable';
import { AppMetrics } from '../../utils/Dimensions';
import Icon from 'react-native-vector-icons/MaterialIcons';
import { styles } from '../../utils/styles';
import { ScrollView } from 'react-native-gesture-handler';
import { IMAGE_BASE_URL } from '../config';

const DetailsScreen = ( { navigation, route } ) =>
{
    const { item, geners } = route.params;
    debugger;
    const buttonRef = useRef();



    return (

        <SafeAreaView>

            <View style={styles.container}>
                {/* <SharedElement id={`item.${ item.id }.image_url`}> */}

                <ImageBackground key={`item.${ item.id }.image_url`}
                    style={{ width: '100%', height: AppMetrics.screenHeight * .4 }} source={{ uri: IMAGE_BASE_URL + item.poster_path }} resizeMode='cover' borderRadius={20} >
                    <Animatable.View
                        key={`item.${ item.id }.view`}
                        ref={buttonRef}
                        animation='fadeIn'
                        duration={600}
                        delay={300}
                        style={[StyleSheet.absoluteFillObject]}
                    >

                        <Icon
                            key={`item.${ item.id }.close`}
                            name='close'
                            size={28}
                            color='#fff'
                            style={{
                                position: 'absolute',
                                top: 40,
                                right: 20,
                                zIndex: 2
                            }}
                            onPress={() =>
                            {
                                buttonRef.current.fadeOut( 100 ).then( () =>
                                {
                                    navigation.goBack();
                                } );
                            }}
                        />
                    </Animatable.View>
                    <View style={styles.languageContainer} key={`item.${ item.id }.langView`}>
                        <Text key={`item.${ item.id }.langText`}
                            style={styles.language}>{item.original_language}</Text>
                    </View>
                </ImageBackground>
                {/* </SharedElement> */}
                <ScrollView showsVerticalScrollIndicator={false}>
                    <View style={{ flexDirection: 'column' }}>
                        {/* <SharedElement id={`item.${ item.id }.cardInfo`}> */}
                        <View key={`item.${ item.id }.infoView`}
                            style={{ width: '100%', justifyContent: 'space-between', flexDirection: 'row', padding: 20 }}>
                            <View style={{ flex: 4 }}>
                                <Text
                                    key={`item.${ item.id }.infoTitle`}
                                    style={styles.title}>{item.title}</Text>
                                <Text key={`item.${ item.id }.infoRealese`}
                                    style={styles.releaseDate}
                                >{item.release_date}</Text>
                            </View>
                            <View style={{ flex: 2, marginHorizontal: 10 }}>
                                <Text key={`item.${ item.id }.infoVotes`}
                                    style={styles.votes}>{item.vote_average}<Text style={styles.votes}>/{item.vote_count}</Text></Text>
                            </View>
                        </View>
                        {/* </SharedElement> */}
                        <Text key={`item.${ item.id }.infoOverview`}
                            style={styles.overview}>{item.overview}</Text>
                        <View style={styles.tagContainer}>
                            {geners.map( ( item, key ) => (
                                <View
                                    key={`item.${ item.id }.tagView`}
                                    style={styles.tagItemContainer}>
                                    <Text key={`item.${ item.id }.tagText`}
                                        style={styles.tagItem}>{item}</Text>
                                </View>
                            ) )}

                        </View>
                    </View>
                </ScrollView>
            </View>
        </SafeAreaView>

    );
};



// DetailsScreen.sharedElements = route =>
// {
//     const { item } = route.params;
//     return [
//         {
//             id: `item.${ item.id }.image_url`,
//             animation: 'move',
//             resize: 'clip'
//         },
//         {
//             id: `item.${ item.id }.cardInfo`,
//             animation: 'fade',
//             resize: 'clip'
//         }
//     ];
// };
export default DetailsScreen;
