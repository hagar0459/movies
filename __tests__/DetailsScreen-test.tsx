/**
 * @format
 */

import 'react-native';
import React from 'react';
import DetailsScreen from '../src/screens/DetailsScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
describe( '<DetailsScreen /> tests', () =>
{
  it( 'Render the component correctly ', () =>
  {

    const route = {
      params: {
        item: {
          adult: false,
          backdrop_path: "/eeijXm3553xvuFbkPFkDG6CLCbQ.jpg",
          genre_ids: [28, 12, 18, 14, 878],
          id: 438631,
          original_language: "en",
          original_title: "Dune",
          overview: "Paul Atreides, a brilliant and gifted young man born into a great destiny beyond his understanding, must travel to the most dangerous planet in the universe to ensure the future of his family and his people. As malevolent forces explode into conflict over the planet's exclusive supply of the most precious resource in existence-a commodity capable of unlocking humanity's greatest potential-only those who can conquer their fear will survive.",
          popularity: 5449.083,
          poster_path: "/d5NXSklXo0qyIYkgV94XAgMIckC.jpg",
          release_date: "2021-09-15",
          title: "Dune",
          video: false,
          vote_average: 8.1,
          vote_count: 2176
        }, geners: ["Action", "Adventure", "Drama", "Fantasy", "Science Fiction"]
      }
    }
    const navigate = jest.fn();
    const tree = renderer.create( <DetailsScreen route={route} navigation={{ navigate }} /> ).toJSON();
    expect( tree ).toMatchSnapshot();
  } );
} );