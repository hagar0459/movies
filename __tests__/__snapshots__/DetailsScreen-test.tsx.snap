// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsScreen /> tests Render the component correctly  1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "height": 533.6,
          "width": "100%",
        }
      }
    >
      <Image
        borderRadius={20}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://www.themoviedb.org/t/p/w440_and_h660_face/d5NXSklXo0qyIYkgV94XAgMIckC.jpg",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 533.6,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
      <View
        collapsable={false}
        forwardedRef={[Function]}
        nativeID="animatedComponent"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <Text
          allowFontScaling={false}
          onPress={[Function]}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 28,
              },
              Object {
                "position": "absolute",
                "right": 20,
                "top": 40,
                "zIndex": 2,
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Óóç
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 20,
            "bottom": 10,
            "position": "absolute",
            "right": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "red",
              "fontSize": 15,
              "fontWeight": "600",
              "margin": 10,
              "textTransform": "uppercase",
            }
          }
        >
          en
        </Text>
      </View>
    </View>
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 20,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 17,
                  }
                }
              >
                Dune
              </Text>
              <Text
                style={
                  Object {
                    "color": "grey",
                    "fontSize": 15,
                  }
                }
              >
                2021-09-15
              </Text>
            </View>
            <View
              style={
                Object {
                  "flex": 2,
                  "marginHorizontal": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignItems": "center",
                    "fontSize": 20,
                    "justifyContent": "center",
                    "textAlign": "center",
                  }
                }
              >
                8.1
                <Text
                  style={
                    Object {
                      "alignItems": "center",
                      "fontSize": 20,
                      "justifyContent": "center",
                      "textAlign": "center",
                    }
                  }
                >
                  /
                  2176
                </Text>
              </Text>
            </View>
          </View>
          <Text
            style={
              Object {
                "fontSize": 17,
                "padding": 20,
              }
            }
          >
            Paul Atreides, a brilliant and gifted young man born into a great destiny beyond his understanding, must travel to the most dangerous planet in the universe to ensure the future of his family and his people. As malevolent forces explode into conflict over the planet's exclusive supply of the most precious resource in existence-a commodity capable of unlocking humanity's greatest potential-only those who can conquer their fear will survive.
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
                "flexWrap": "wrap",
                "marginHorizontal": 20,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 15,
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "margin": 7,
                  }
                }
              >
                Action
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 15,
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "margin": 7,
                  }
                }
              >
                Adventure
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 15,
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "margin": 7,
                  }
                }
              >
                Drama
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 15,
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "margin": 7,
                  }
                }
              >
                Fantasy
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "grey",
                  "borderRadius": 15,
                  "marginBottom": 5,
                  "marginLeft": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 14,
                    "margin": 7,
                  }
                }
              >
                Science Fiction
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
